### PR TITLE
update targets

### PR DESCRIPTION
**Pull request by meeuw**
#### Which issue(s) does this change fix?

N/A

#### Why is this change necessary?

Makefile, Make.ps1 and appveyor.yml are inconsistent, it should be managed from a centralized place.

#### How does it address the issue?

By generating Makefile, Make.ps1 and appveyor.yml from templates and a central definition (targets.yaml) of the targets.

#### What side effects does this change have?

It increases the total amount of code in this repository.

#### Checklist

- [x] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- ~Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))~
- ~Write unit tests~
- ~Write/update functional tests~
- ~Write/update integration tests~
- [ ] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [x] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

**Comment by meeuw** [comment]
@ssenchenko as you added Make.ps1 in https://github.com/aws/aws-sam-cli/pull/3179 could you please have a look?

File: Make.ps1 [hunk 1]
```diff
@@ -16,94 +20,142 @@
...
-    # $UpdateReproducableReqs
+    # Run unit tests with html coverage report
+    [Parameter(ParameterSetName = 'TestCovReport')]
+    [switch]
+    $TestCovReport,
+
+    # Update reproducable requirements.
+    [Parameter(ParameterSetName = 'UpdateReproducibleReqs')]
+    [switch]
+    $UpdateReproducibleReqs,
```
Comment (meeuw) [inline Make.ps1:88]:
```suggestion
    $UpdateReproducibleReqs
```

It's not allowed to have a trailing `,` after the last item

File: Make.ps1 [hunk 2]
```diff
@@ -16,94 +20,142 @@
...
 )
 
+
+function Black {
+    black setup.py samcli tests
```
Comment (meeuw) [inline Make.ps1:93]: This seems to cause an infinitive recursion (black calling black).

File: Make.ps1 [hunk 3]
```diff
@@ -16,94 +20,142 @@
...
+}
+
+function Dev {
+    Lint
+    Test
+}
+
+function FuncTest {
+    # Verify function test coverage only for `samcli.local` package
+    @echo Telemetry Status: $[SAM_CLI_TELEMETRY]
```
Comment (meeuw) [inline Make.ps1:107]:
```suggestion
    echo Telemetry Status: $[SAM_CLI_TELEMETRY]
```
`@echo` isn't allowed

**Comment by ssenchenko** [comment]
@meeuw Thank you for your suggestion. Overall, your idea sounds good. I'll certainly look into details in a couple of days.

**Comment by jfuss** [comment]
Closing as this looks inactive and all builds fail.

If you are still wanting to contribute this, please feel free to re-open or submit a new PR and have the build passing so we can review.

**Pull request closed without merging**